src/AppBundle/Entity/Post.php

```diff
@@ -133,13 +133,13 @@ public function setAuthorEmail($authorEmail)
     /**
      * Is the given User the author of this Post?
      *
-     * @param User $user
+     * @param User|string $user
```

Comment:
in any case, only a User object should reach this method call. If you have places where you don't have a User, it should be handled in the code calling this method, not in the Entity. For the entity, it makes perfect sense to expect a User object